scipy/stats/morestats.py

```diff
@@ -1329,7 +1329,7 @@ def yeojohnson(x, lmbda=None):
 
     """
 
-    x = np.asarray(x)
+    x = np.asarray(x, dtype=np.float64)
```

Comment:
As you note in the issue, some consideration for passing through smaller dtypes might be warranted. Would people typically call this function with a smaller float type and expect the return to be of the same type? The promotion logic for doing that and supporting integers might be a little messy maybe.